src/peft/tuners/adalora/model.py

```diff
@@ -349,3 +349,7 @@ def update_and_allocate(self, global_step):
         # Pass the function and do forward propagation
         else:
             return None
+
+    def add_weighted_adapter(self, *args, **kwargs):
```

Comment:
Nit: we could maybe upstream that into the base model class

Reply:
Not sure what you mean, could you please explain this.

Reply:
i meant we could add this method in the basetuner class

Reply:
So as is, if the adapter type is not LoRA, this method does not exist. If we move it to the `BaseTuner`, it would be there but raise an error, I think it's more confusing than helping. And we would still need it here because `AdaLoraModel` inherits from `LoraModel`.

Reply:
Makes sense!